query-algebrizer/src/clauses/mod.rs

```diff
@@ -89,6 +89,7 @@ trait Contains<K, T> {
 
 trait Intersection<K> {
     fn with_intersected_keys(&self, ks: &BTreeSet<K>) -> Self;
+    fn keep_intersected_keys(&mut self, ks: &BTreeSet<K>);
```

Comment:
Well, it means we can write:

```
            value_bindings: self.value_bindings.with_intersected_keys(&vars),
```

rather than

```
let new_value_bindings = self.value_bindings.clone();
new_value_bindings.keep_intersected_keys(&vars);
…
    value_bindings: new_value_bindings,
```

The reason for `k_i_k` is because we're already taking ownership of the inputs, so it's wasteful to clone into a new map.